src/librustdoc/html/static/main.js

```diff
@@ -428,9 +428,9 @@ function hideThemeButtonState() {
     function handleEscape(ev) {
         var help = getHelpElement(false);
         var search = searchState.outputElement();
-        if (!hasClass(help, "hidden")) {
+        if (help && !hasClass(help, "hidden")) {
```

Comment:
hasClass is defined like so:

```js
function hasClass(elem, className) {
    return elem && elem.classList && elem.classList.contains(className);
}
```

So I think this should be redundant. Can you explain (in the PR description) what the bug was and how this fixes?

Reply:
It's simply because if the element doesn't exist, it doesn't have the class and enter the condition. :)

Reply:
What about other cases? I don't see the first `help && ` part, some of them have this and some does not. Seemed confusing to me.

Reply:
Aha, now I see the issue. The problem is that these `if` clauses are exclusive. So we were mistakenly executing the "hide help" clause, which prevented us from executing the "hide search" clause.

What about making both of these clauses unconditional? If you hit escape, it's reasonable to try and hide the help, and also try and hide the search. Worst case, it's a no-op.

Relatedly: Why does only the search clause have a preventDefault()? Do we need a preventDefault() at all here?

Reply:
Well, we might want to close the help without closing the search results (my GUI test actually enforces this behaviour). Also, this change is mostly because of the migration of the boolean comparison. With `=== false`, it wouldn't enter because `hasClass` would return `null`.

As for `preventDefault`, no idea. Can be removed I think.

Reply:
Now I fully understand. Thanks for explaining.

Reply:
@pickfire Well, not really, in case we want to check that if the element exists, it's hidden, if it doesn't exist then it's all good. You can see it as follows too: if the element doesn't exist, it doesn't implement the class. :smiley: 

@jsha I'll open an issue to make another cleanup round around `hasClass` usage just in case.